src/main/java/com/internet/shop/web/filters/AuthorizationFilter.java

```diff
@@ -0,0 +1,74 @@
+package com.internet.shop.web.filters;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Role;
+import com.internet.shop.model.User;
+import com.internet.shop.service.UserService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthorizationFilter implements Filter {
+    private static final String USER_ID = "user_Id";
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private UserService userService = (UserService) injector
+            .getInstance(UserService.class);
+    private Map<String, List<Role.RoleName>> protectedUrl = new HashMap<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        protectedUrl.put("/products/add", List.of(Role.RoleName.ADMIN));
+        protectedUrl.put("/users", List.of(Role.RoleName.ADMIN));
+        protectedUrl.put("/admin/products", List.of(Role.RoleName.ADMIN));
+        protectedUrl.put("/admin/orders", List.of(Role.RoleName.ADMIN));
+        protectedUrl.put("/admin", List.of(Role.RoleName.ADMIN));
+        protectedUrl.put("/products", List.of(Role.RoleName.USER));
+        protectedUrl.put("/shoppingCart/products", List.of(Role.RoleName.USER));
+        protectedUrl.put("/user/orders", List.of(Role.RoleName.USER));
+        protectedUrl.put("/user", List.of(Role.RoleName.USER));
```

Comment:
complete order, add and remove from shopping cart

Reply:
done